src/org/exist/xquery/functions/fn/FunSort.java

```diff
@@ -0,0 +1,223 @@
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.xquery.functions.fn;
+
+import java.text.Collator;
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.exist.dom.QName;
+import org.exist.util.Collations;
+import org.exist.xquery.AnalyzeContextInfo;
+import org.exist.xquery.BasicFunction;
+import org.exist.xquery.Cardinality;
+import org.exist.xquery.Constants;
+import org.exist.xquery.ErrorCodes;
+import org.exist.xquery.Function;
+import org.exist.xquery.FunctionSignature;
+import org.exist.xquery.XPathException;
+import org.exist.xquery.XQueryContext;
+import org.exist.xquery.value.FunctionParameterSequenceType;
+import org.exist.xquery.value.FunctionReference;
+import org.exist.xquery.value.FunctionReturnSequenceType;
+import org.exist.xquery.value.Item;
+import org.exist.xquery.value.Sequence;
+import org.exist.xquery.value.SequenceIterator;
+import org.exist.xquery.value.SequenceType;
+import org.exist.xquery.value.Type;
+import org.exist.xquery.value.ValueSequence;
+
+public class FunSort extends BasicFunction {
+
+  public final static FunctionSignature signatures[] = {
+    new FunctionSignature(
+      new QName("sort", Function.BUILTIN_FUNCTION_NS),
+        "Sorts a supplied sequence.",
+      new SequenceType[] {
+        new FunctionParameterSequenceType("input", Type.ITEM, Cardinality.ZERO_OR_MORE, "")
+      },
+      new FunctionReturnSequenceType(Type.ITEM, Cardinality.ZERO_OR_MORE, "the first item or the empty sequence")
+    ),
+    new FunctionSignature(
+      new QName("sort", Function.BUILTIN_FUNCTION_NS),
+        "Sorts a supplied sequence, based on the value of a sort key supplied as a function.",
+      new SequenceType[] {
+          new FunctionParameterSequenceType("input", Type.ITEM, Cardinality.ZERO_OR_MORE, ""),
+          new FunctionParameterSequenceType("collation", Type.STRING, Cardinality.ZERO_OR_ONE, "")
+      },
+      new FunctionReturnSequenceType(Type.ITEM, Cardinality.ZERO_OR_MORE, "the resulting sequence")
+    ),
+    new FunctionSignature(
+        new QName("sort", Function.BUILTIN_FUNCTION_NS),
+        "Sorts a supplied sequence, based on the value of a sort key supplied as a function.",
+        new SequenceType[] {
+            new FunctionParameterSequenceType("input", Type.ITEM, Cardinality.ZERO_OR_MORE, ""),
+            new FunctionParameterSequenceType("collation", Type.STRING, Cardinality.ZERO_OR_ONE, ""),
+            new FunctionParameterSequenceType("key", Type.FUNCTION_REFERENCE, Cardinality.ONE, "")
+        },
+        new FunctionReturnSequenceType(Type.ITEM, Cardinality.ZERO_OR_MORE, "the resulting sequence")
+    )
+  };
+
+  AnalyzeContextInfo cachedContextInfo;
+
+  public FunSort(XQueryContext context, FunctionSignature signature) {
+    super(context, signature);
+  }
+
+  @Override
+  public void analyze(AnalyzeContextInfo contextInfo) throws XPathException {
+    if (getContext().getXQueryVersion()<30) {
```

Comment:
@dizzzz is correct. `fn:sort()` was added in XQuery 3.1 F&O; see https://www.w3.org/TR/xpath-functions-31/#substantive-changes.